go/internal/feast/sqliteonlinestore.go

```diff
@@ -0,0 +1,159 @@
+	} else {
+		store.path = fmt.Sprintf("%s/%s", repoConfig.RepoPath, dbPathStr)
+		db, err := initializeConnection(store.path)
+		if err != nil {
+			return nil, err
+		}
+		store.db = db
+	}
+	return &store, nil
+}
+
+func (s *SqliteOnlineStore) Destruct() {
+	s.db.Close()
+}
+
+// Returns FeatureData 2D array. Each row corresponds to one entity value and each column corresponds to a single feature where the number of columns should be
+// same length as the length of featureNames. Reads from every table in featureViewNames with the entity keys described.
+func (s *SqliteOnlineStore) OnlineRead(ctx context.Context, entityKeys []types.EntityKey, featureViewNames []string, featureNames []string) ([][]FeatureData, error) {
+	featureCount := len(featureNames)
+	_, err := s.getConnection()
+	if err != nil {
+		return nil, err
+	}
+
+	project := s.project
+	results := make([][]FeatureData, len(entityKeys))
+	entityNameToEntityIndex := make(map[string]int)
+	in_query := make([]string, len(entityKeys))
+	serialized_entities := make([]interface{}, len(entityKeys))
+	for i := 0; i < len(entityKeys); i++ {
+		serKey, err := serializeEntityKey(&entityKeys[i])
+		if err != nil {
+			return nil, err
+		}
+		// TODO: fix this, string conversion is not safe
+		entityNameToEntityIndex[string(*serKey)] = i
+		// for IN clause in read query
+		in_query[i] = "?"
+		serialized_entities[i] = *serKey
+	}
+	featureNamesToIdx := make(map[string]int)
+	for idx, name := range featureNames {
+		featureNamesToIdx[name] = idx
+	}
+
+	for idx := 0; idx < len(entityKeys); idx++ {
+		results[idx] = make([]FeatureData, featureCount)
+	}
+	for _, featureViewName := range featureViewNames {
+		query_string := fmt.Sprintf(`SELECT entity_key, feature_name, value, event_ts
+									FROM %s
+									WHERE entity_key IN (%s)
+									ORDER BY entity_key`, tableId(project, featureViewName), strings.Join(in_query, ","))
+		rows, err := s.db.Query(query_string, serialized_entities...)
+		if err != nil {
+			return nil, err
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var entity_key string
+			var feature_name string
+			var valueString string
+			var event_ts time.Time
+			var value types.Value
+			err = rows.Scan(&entity_key, &feature_name, &valueString, &event_ts)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if err := proto.Unmarshal([]byte(valueString), &value); err != nil {
+				return nil, errors.New("error converting parsed value to types.Value")
+			}
+			results[entityNameToEntityIndex[entity_key]][featureNamesToIdx[feature_name]] = FeatureData{reference: serving.FeatureReferenceV2{FeatureViewName: featureViewName, FeatureName: feature_name},
+				timestamp: *timestamppb.New(event_ts),
+				value:     types.Value{Val: value.Val},
+			}
+		}
+	}
+	return results, nil
+}
+
+func (s *SqliteOnlineStore) Update(ctx context.Context, config *RepoConfig, tables_to_delete []*FeatureView, tables_to_keep []*FeatureView) error {
```

Comment:
Do you need this if we never write rows to the sqlite database?

Reply:
I'll remove it for now.